client/my-sites/checkout/checkout/billing-details-form.jsx

```diff
@@ -0,0 +1,283 @@
+		const newState = {
+			steps: updatedSteps,
+		};
+		if ( updatedSteps.indexOf( this.state.currentStep ) < 0 ) {
+			debug( 'Switching to step: mainForm' );
+			newState.currentStep = 'mainForm';
+		}
+		this.setState( newState );
+	}
+
+	validate = ( fieldValues, onComplete ) => {
+		const validationHandler = ( error, data ) => {
+			const messages = ( data && data.messages ) || {};
+			onComplete( error, messages );
+		};
+
+		if ( this.needsOnlyGoogleAppsDetails() ) {
+			wpcom.validateGoogleAppsContactInformation( fieldValues, validationHandler );
+			return;
+		}
+
+		wpcom.validateDomainContactInformation( fieldValues, this.getDomainNames(), validationHandler );
+	};
+
+	hasAnotherStep() {
+		return this.state.currentStep !== last( this.state.steps );
+	}
+
+	switchToNextStep() {
+		const newStep = this.state.steps[ indexOf( this.state.steps, this.state.currentStep ) + 1 ];
+		debug( 'Switching to step: ' + newStep );
+		this.setState( { currentStep: newStep } );
+	}
+
+	needsDomainDetails() {
+		const cart = this.props.cart;
+
+		if ( cart && cartItems.hasOnlyRenewalItems( cart ) ) {
+			return false;
+		}
+
+		return (
+			cart &&
+			( cartItems.hasDomainRegistration( cart ) ||
+				cartItems.hasGoogleApps( cart ) ||
+				cartItems.hasTransferProduct( cart ) )
+		);
+	}
+
+	getDomainNames = () =>
+		map(
+			[
+				...cartItems.getDomainRegistrations( this.props.cart ),
+				...cartItems.getDomainTransfers( this.props.cart ),
+			],
+			'meta'
+		);
+
+	needsOnlyGoogleAppsDetails() {
+		return (
+			cartItems.hasGoogleApps( this.props.cart ) &&
+			! cartItems.hasDomainRegistration( this.props.cart ) &&
+			! cartItems.hasTransferProduct( this.props.cart )
+		);
+	}
+
+	getNumberOfDomainRegistrations() {
+		return cartItems.getDomainRegistrations( this.props.cart ).length;
+	}
+
+	getTldsWithAdditionalForm() {
+		if ( ! config.isEnabled( 'domains/cctlds' ) ) {
+			// All we need to do to disable everything is not show the .FR form
+			return [];
+		}
+		return intersection( cartItems.getTlds( this.props.cart ), tldsWithAdditionalDetailsForms );
+	}
+
+	needsFax() {
+		return (
+			this.props.contactDetails.countryCode === 'NL' && cartItems.hasTld( this.props.cart, 'nl' )
+		);
+	}
+
+	getSubmitButtonText() {
+		return this.hasAnotherStep()
+			? this.props.translate( 'Continue' )
+			: this.props.translate( 'Continue to Checkout' );
+	}
+
+	renderSubmitButton() {
+		return (
+			<FormButton
+				className="checkout__domain-details-form-submit-button"
+				onClick={ this.handleSubmitButtonClick }
+			>
+				{ this.getSubmitButtonText() }
+			</FormButton>
+		);
+	}
+
+	handleContactDetailsChange = newContactDetailsValues => {
+		this.props.updateContactDetailsCache( newContactDetailsValues );
+	};
+
+	renderBillingContactDetailsFields() {
+		const { contactDetails, translate, userCountryCode } = this.props;
+		const labelTexts = {
+			organization: translate( '+ Add Organization Name' ),
```

Comment:
Hi! I've found a possible matching string that has already been translated **19** times:
`translate( '+ Add organization name' )` *ES Score: 19*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/billing-info-to-checkout)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).